### PR TITLE
fix: update s3 bucket policies to adhere to best practices

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes
Adds BPA config to the deployment bucket and also attaches a bucket policy that blocks all unencrypted traffic to the bucket. This is to adhere to best practices for private buckets.

Also fixes some lint errors in the file

#### Description of how you validated changes
Tested that this does not cause any interruption to existing projects. Also tested that init still works properly. Updated existing snapshot tests.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by johnpc — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts [hunk 1]
```diff
@@ -140,51 +173,67 @@ export class AmplifyRootStack extends cdk.Stack implements AmplifyRootStackTempl
...
+  addCfnMapping(): void {
     throw new Error('Method not implemented.');
   }
-  addCfnCondition(props: cdk.CfnConditionProps, logicalId: string): void {
+
+  /**
+   * Method not implemented
+   */
+  // eslint-disable-next-line class-methods-use-this
+  addCfnCondition(/* _props: cdk.CfnConditionProps, _logicalId: string */): void {
```
Comment (sachscode) [inline packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts:225]: was this added intentionally ?

File: packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts [hunk 2]
```diff
@@ -140,51 +173,67 @@ export class AmplifyRootStack extends cdk.Stack implements AmplifyRootStackTempl
...
     throw new Error('Method not implemented.');
   }
-  addCfnResource(props: cdk.CfnResourceProps, logicalId: string): void {
+
+  /**
+   * Method not implemented
+   */
+  // eslint-disable-next-line class-methods-use-this
+  addCfnResource(/* _props: cdk.CfnResourceProps, _logicalId: string */): void {
```
Comment (sachscode) [inline packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts:233]: was this added intentionally ?

File: packages/amplify-provider-awscloudformation/src/__tests__/root-stack-builder/__snapshots__/root-stack-builder.test.ts.snap [hunk 3]
```diff
@@ -100,10 +100,63 @@ Object {
...
+          "Statement": Array [
+            Object {
+              "Action": "s3:*",
+              "Condition": Object {
+                "Bool": Object {
+                  "aws:SecureTransport": false,
+                },
+              },
+              "Effect": "Deny",
+              "Principal": "*",
```
Comment (sachscode) [inline packages/amplify-provider-awscloudformation/src/__tests__/root-stack-builder/__snapshots__/root-stack-builder.test.ts.snap:128]: Will deployment work in this case ? The IAM role used to deploy the project would need write permissions.

**Review by sachscode — COMMENTED** [review]
LGTM , just a few questions

File: packages/amplify-provider-awscloudformation/src/__tests__/root-stack-builder/__snapshots__/root-stack-builder.test.ts.snap (hunk 3 above)
Comment (edwardfoyle) [reply packages/amplify-provider-awscloudformation/src/__tests__/root-stack-builder/__snapshots__/root-stack-builder.test.ts.snap:128]: This only denies unsecured traffic to the bucket. Any connection using HTTPS goes through

File: packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts (hunk 1 above)
Comment (edwardfoyle) [reply packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts:225]: this was to get the linter to stop complaining about unused parameters but still show what the parameters should be if/when this method needs to be implemented

**Review by sachscode — APPROVED** [review]
LGTM

**Pull request merged**